src/compiler/filesystem_local.cc

```diff
@@ -36,7 +36,11 @@ extern int __xstat64(int ver, const char* path, struct stat64* stat_buf);
 
 }
 
-#ifdef BUILD_64
+#if !defined(__riscv) && !defined(TOIT_LINUX)
+  #define USE_XSTAT64 1
+#endif
+
+#ifndef BUILD_64
```

Comment:
Was this a bug before?

Reply:
Both xstat versions 1 and 3 returned error code -1 on riscv64, but stat works fine. 

riscv64/Linux builds are still Linux, so it seemed like a good idea to use a more descriptive define check.

Reply:
I was thinking more about the change from 

```
ifdef BUILD_64
```

to

```
ifndef BUILD_64
```